test/jdk/java/util/Collections/Shuffle.java

```diff
@@ -0,0 +1,93 @@
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug     8294693
+ * @summary Basic test for Collections.shuffle
+ */
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Random;
+import java.util.random.RandomGenerator;
+import java.util.random.RandomGeneratorFactory;
+
+public class Shuffle {
+    static final int N = 100;
+
+    public static void main(String[] args) {
+        test(new ArrayList<>());
+        test(new LinkedList<>());
+    }
+
+    static void test(List<Integer> list) {
+        for (int i = 0; i < N; i++) {
+            list.add(i);
+        }
+        Collections.shuffle(list);
+        if (list.size() != N) {
+            throw new RuntimeException(list.getClass() + ": size " + list.size() + " != " + N);
+        }
+        for (int i = 0; i < N; i++) {
+            if (!list.contains(i)) {
+                throw new RuntimeException(list.getClass() + ": does not contain " + i);
+            }
+        }
+        checkRandom(list);
+        checkRandomGenerator(list);
+    }
+
+    private static void checkRandomGenerator(List<Integer> list) {
+        RandomGeneratorFactory<RandomGenerator> factory = RandomGeneratorFactory.getDefault();
+        list.sort(null);
+        Collections.shuffle(list, factory.create(1));
+        ArrayList<Integer> copy = new ArrayList<>(list);
+        list.sort(null);
+        if (list.equals(copy)) {
+            // Assume that at least one pair of elements must be reordered during shuffle
+            throw new RuntimeException(list.getClass() + ": list is not shuffled");
+        }
+        Collections.shuffle(list, factory.create(1));
+        if (!list.equals(copy)) {
+            throw new RuntimeException(list.getClass() + ": " + list + " != " + copy);
+        }
+    }
+
+    private static void checkRandom(List<Integer> list) {
+        list.sort(null);
+        Collections.shuffle(list, new Random(1));
+        ArrayList<Integer> copy = new ArrayList<>(list);
+        list.sort(null);
+        if (list.equals(copy)) {
+            // Assume that at least one pair of elements must be reordered during shuffle
+            throw new RuntimeException(list.getClass() + ": list is not shuffled");
+        }
+        Collections.shuffle(list, new Random(1));
+        if (!list.equals(copy)) {
+            throw new RuntimeException(list.getClass() + ": " + list + " != " + copy);
+        }
+    }
```

Comment:
Is there a way to factor out the `shuffle` calls and thereby collapse these two methods into one? Is it worth it? I'm thinking you could pass in a `Consumer<List<Integer>>`.